### PR TITLE
Feat/explorer scheduled deletion

### PR DESCRIPTION
**Pull request by edge33**
### Ticket:

[User Story 18797](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/18797): delete files

### Description

Added scheduled deletion for files when removed

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Dev change**
- [ ] **Additional tests**
- [ ] **Documentation**
- [ ] **Other**

### Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] This change might impact the developer experience of others and should be communicated

### Screenshots:

N/A

### CODE REVIEW

File: apps/upload-api/e2e/PublicServant/Share/Create file sharing.bru [hunk 1]
```diff
@@ -0,0 +1,25 @@
+meta {
+  name: Create file sharing
+  type: http
+  seq: 1
+}
+
+post {
+  url: {{baseUrl}}/api/v1/metadata/share
+  body: formUrlEncoded
+  auth: none
```
Comment (Thurfjell) [inline apps/upload-api/e2e/PublicServant/Share/Create file sharing.bru:10]: Don't you need `auth: bearer` ?

File: apps/upload-api/src/routes/files/index.ts [hunk 2]
```diff
@@ -211,54 +209,54 @@ export default async function routes(app: FastifyInstance) {
...
-
-        await deleteFileMetadata(app.pg, fileId);
-      } catch (err) {
-        throw new ServerError(FILE_DELETE, "Internal server error", err);
-      }
-
-      reply.send({ data: { message: "File deleted succesfully" } });
-    },
-  );
+  // app.delete<{ Params: { id: string } }>(
```
Comment (Thurfjell) [inline apps/upload-api/src/routes/files/index.ts:212]: Saving for later?

**Review by Thurfjell — COMMENTED** [review]
You're not using the scheduler right?

File: apps/upload-api/e2e/PublicServant/Share/Create file sharing.bru (hunk 1 above)
Comment (edge33) [reply apps/upload-api/e2e/PublicServant/Share/Create file sharing.bru:10]: I do have it 🤔

File: apps/upload-api/src/routes/files/index.ts (hunk 2 above)
Comment (edge33) [reply apps/upload-api/src/routes/files/index.ts:212]: indeed, I need this code as reference for when I start the scheduler work

File: apps/upload-api/e2e/PublicServant/Share/Create file sharing.bru (hunk 1 above)
Comment (Thurfjell) [reply apps/upload-api/e2e/PublicServant/Share/Create file sharing.bru:10]: Ah you set in header. Alright fair, you can also do like

```
post {
  url: blublub
  body: none
  auth: bearer
}

auth:bearer {
  token: {{accessToken}}
}
```

**Review by Thurfjell — APPROVED** [review]

**Pull request merged**